core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java

```diff
@@ -43,110 +52,239 @@
+                }
+            },
+            RLE_BIGINT(BigintType.BIGINT, 3000) {
+                @Override
+                public Page createPage(List<Type> types, int positionCount, float primitiveNullRate, float nestedNullRate)
+                {
+                    return createRlePageWithRandomData(types, POSITION_COUNT, primitiveNullRate, nestedNullRate);
+                }
+            },
+            LONG_DECIMAL(createDecimalType(MAX_SHORT_PRECISION + 1), 5000),
+            INTEGER(IntegerType.INTEGER, 5000),
+            SMALLINT(SmallintType.SMALLINT, 5000),
+            BOOLEAN(BooleanType.BOOLEAN, 5000),
+            VARCHAR(VarcharType.VARCHAR, 5000),
+            ARRAY_BIGINT(new ArrayType(BigintType.BIGINT), 1000),
+            ARRAY_VARCHAR(new ArrayType(VarcharType.VARCHAR), 1000),
+            ARRAY_ARRAY_BIGINT(new ArrayType(new ArrayType(BigintType.BIGINT)), 1000),
+            MAP_BIGINT_BIGINT(createMapType(BigintType.BIGINT, BigintType.BIGINT), 1000),
+            MAP_BIGINT_MAP_BIGINT_BIGINT(createMapType(BigintType.BIGINT, createMapType(BigintType.BIGINT, BigintType.BIGINT)), 1000),
+            ROW_BIGINT_BIGINT(withDefaultFieldNames(ImmutableList.of(BigintType.BIGINT, BigintType.BIGINT)), 1000),
+            ROW_ARRAY_BIGINT_ARRAY_BIGINT(withDefaultFieldNames(ImmutableList.of(new ArrayType(BigintType.BIGINT), new ArrayType(BigintType.BIGINT))), 1000);
+
+            private final Type type;
+            private final int pageCount;
+
+            TestType(Type type, int pageCount)
+            {
+                this.type = type;
+                this.pageCount = pageCount;
+            }
+
+            public Page createPage(List<Type> types, int positionCount, float primitiveNullRate, float nestedNullRate)
+            {
+                return createPageWithRandomData(types, POSITION_COUNT, primitiveNullRate, nestedNullRate);
+            }
+
+            public Type type()
+            {
+                return type;
+            }
+
+            public int pageCount()
+            {
+                return pageCount;
+            }
+        }
+
+        @Setup
+        public void setup()
+        {
+            createPages(type);
+        }
+
+        private void createPages(TestType inputType)
+        {
+            float primitiveNullRate = 0.0f;
+            float nestedNullRate = 0.0f;
+
+            if (hasNull) {
+                primitiveNullRate = 0.2f;
+                nestedNullRate = 0.2f;
+            }
+            types = nCopies(channelCount, inputType.type());
+            dataPage = inputType.createPage(types, POSITION_COUNT, primitiveNullRate, nestedNullRate);
+            pageCount = inputType.pageCount();
+
+            // We built the dataPage with added pre-computed hash block at channel 0, so types needs to be updated
+            types = createUpdatedBlockTypesWithHashBlockAndNullBlock(types, true, false);
```

Comment:
just add a block here and remove `createUpdatedBlockTypesWithHashBlockAndNullBlock` method